### DISCUSSION TITLE
Use SWC as transpiler

### DISCUSSION THREAD

**Discussion by vjpr** [oven-sh/bun#1177] [Ideas]
SWC has a great wasm plugin api. esbuild's plugin system is more limited.

**Comment by Jarred-Sumner** [comment]
You can use SWC's wasm api if you'd like, but additional usage is outside the scope of Bun. Bun uses bun's transpiler extensively in the runtime and it would be very complicated to add others builtin. 

Are there specific features missing from Bun's transpiler that you need?

**Comment by vjpr** [comment]
Vite 4 now supports SWC as an option for React projects.

There seems to be a lot of projects (Next.js, Turbopack, [Deno](https://www.reddit.com/r/rust/comments/i02zu3/deno_now_uses_swc_a_tsjs_compiler_in_rust_for/), Rome) and companies coaelescing around SWC and Rust-based tooling . I wouldn't be surprised if `esbuild` was sunset sometime in the future. It's a solo dev effort.

Might be worth thinking more about supporting SWC for future-proofing.

## Details

https://vitejs.dev/blog/announcing-vite4.html

> From Vite 4, two plugins are available for React projects with different tradeoffs. We believe that both approaches are worth supporting at this point, and we'll continue to explore improvements to both plugins in the future.

> @vitejs/plugin-react-swc is a new plugin that uses esbuild during build, but replaces Babel with SWC during development.

**Comment by Jarred-Sumner** [comment]
If there are no specific features or bugs then I'm going to go ahead and lock this because the question has been answered
